explorer/nodewatch/tests/model/test_NodeInfo.py

```diff
@@ -0,0 +1,176 @@
+		self.assertEqual('http://mysymbolnode.com:3000', info.endpoint)
+		self.assertEqual('xym rocks', info.name)
+		self.assertEqual('1.0.3.5', info.version)
+		self.assertEqual(5, info.roles)
+
+	def test_has_api_returns_false_when_api_role_is_not_set(self):
+		for role in (1, 4, 5, 0xFD):
+			# Arrange:
+			info = self._create_default_info(('roles', role))
+
+			# Act + Assert:
+			self.assertFalse(info.has_api)
+
+	def test_has_api_returns_true_when_api_role_is_set(self):
+		for role in (1, 4, 5, 0xFD):
+			# Arrange:
+			info = self._create_default_info(('roles', role | NodeInfo.API_ROLE_FLAG))
+
+			# Act + Assert:
+			self.assertTrue(info.has_api)
+
+	def test_can_convert_to_simple_dict(self):
+		# Arrange:
+		info = self._create_default_info()
+
+		# Act:
+		info_dict = info.to_dict()
+
+		# Assert:
+		self.assertEqual({
+			'mainPublicKey': 'FDA457356DEB75DFBB7FC3B1B29E8A82ECE510CB4A471C1E0DCC9C889F2648C2',
+			'nodePublicKey': '7663D638DDB64CEA9CD9FBFD9AD43B41D0CBA0141117E43243886BA6E8109912',
+			'endpoint': 'http://mysymbolnode.com:3000',
+			'name': 'xym rocks',
+			'version': '1.0.3.5',
+			'roles': 5
+		}, info_dict)
+
+	def test_can_convert_to_simple_dict_when_node_public_key_is_missing(self):
+		# Arrange:
+		info = self._create_default_info()
+		info.node_public_key = None
+
+		# Act:
+		info_dict = info.to_dict()
+
+		# Assert:
+		self.assertEqual({
+			'mainPublicKey': 'FDA457356DEB75DFBB7FC3B1B29E8A82ECE510CB4A471C1E0DCC9C889F2648C2',
+			'nodePublicKey': None,
+			'endpoint': 'http://mysymbolnode.com:3000',
+			'name': 'xym rocks',
+			'version': '1.0.3.5',
+			'roles': 5
+		}, info_dict)
+
+	def test_eq_is_supported(self):
+		# Arrange:
+		info = self._create_default_info()
+		other_public_key = PublicKey('C8B16FB70B85EBC341E2D71DFFDCE1E8020018B28637EB50E82057ACB039AFB1')
+		other_hash = Hash256('C8B16FB70B85EBC341E2D71DFFDCE1E8020018B28637EB50E82057ACB039AFB1')
+
+		# Act + Assert:
+		self.assertEqual(info, self._create_default_info())
+		self.assertNotEqual(info, None)
+		self.assertNotEqual(info, 17)
+		self.assertNotEqual(info, self._create_default_info(('network_id', 111)))
+		self.assertNotEqual(info, self._create_default_info(('network_generation_hash_seed', other_hash)))
+		self.assertNotEqual(info, self._create_default_info(('main_public_key', other_public_key)))
+		self.assertNotEqual(info, self._create_default_info(('node_public_key', other_public_key)))
+		self.assertNotEqual(info, self._create_default_info(('endpoint', 'https://mysymbolnode.com:3001')))
+		self.assertNotEqual(info, self._create_default_info(('name', 'xym xym')))
+		self.assertNotEqual(info, self._create_default_info(('version', '1.0.3.4')))
+		self.assertNotEqual(info, self._create_default_info(('roles', '4')))
+
+	def test_repr_is_supported(self):
+		# Arrange:
+		info = self._create_default_info()
+
+		# Act:
+		info_repr = repr(info)
+		info2 = eval(info_repr)  # pylint: disable=eval-used
+
+		# Assert:
+		self.assertEqual('\n'.join([
```

Comment:
hmm, wouldn't it have more sense to just wrap expected thing in multi-line string `"""`?

Reply:
no real preference, but i generally find this a bit more readable, as the ''' string would need to be deindented relative to the code